go/arrow/array/list_test.go

```diff
@@ -211,3 +211,190 @@ func TestListArraySlice(t *testing.T) {
 		t.Fatalf("got=%q, want=%q", got, want)
 	}
 }
+
+func TestLargeListArray(t *testing.T) {
+	pool := memory.NewCheckedAllocator(memory.NewGoAllocator())
+	defer pool.AssertSize(t, 0)
+
+	var (
+		vs      = []int32{0, 1, 2, 3, 4, 5, 6}
+		lengths = []int{3, 0, 4}
+		isValid = []bool{true, false, true}
+		offsets = []int64{0, 3, 3, 7}
+	)
+
+	lb := array.NewLargeListBuilder(pool, arrow.PrimitiveTypes.Int32)
+	defer lb.Release()
+
+	for i := 0; i < 10; i++ {
```

Comment:
Just for my understanding, why are we looping 10 times over the test body below? Is it to try and detect memory management issues?

Reply:
> is it to try and detect memory management issues?

Basically yes. Calling `NewArray()` should make sure the builder is properly reset to enable reuse (along with the child builders) without any memory leaks. So we loop multiple times, building and calling NewArray to ensure that there are no memory management issues

Reply:
Still for my curiosity, does the Go ecosystem have equivalents to Address Sanitizer in C++, and/or do you Valgrind?

Reply:
Yep! there's a `-asan` option you can pass when building, and the `-race` argument in our CI that we use when running the tests finds race conditions. 

I don't think there's a *full* valgrind equivalent for Go, though there's lots of great memory profiling tools. I can add the `-asan` argument to the CI scripts if you like so that we run all the tests using asan.

Reply:
I think that can be nice if it doesn't make build times too large.

Reply:
Darn! the `-asan` option was introduced with Go1.18, so we'd need to make a new build job running with the newer Go version to leverage it in the CI, until we upgrade the Go version we're using. *sigh*

Reply:
I see, perhaps add a JIRA for an additional asan job?